### PR TITLE
Moved many operations off array/set expressions to apply functions.

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/docs/expr/functions.rst [hunk 1]
```diff
@@ -69,60 +130,113 @@ Functions
...
 .. autofunction:: ppois
 .. autofunction:: qchisqtail
 .. autofunction:: qnorm
 .. autofunction:: qpois
-.. autofunction:: range
+
+Collection constructors
+-----------------------
+
+.. autofunction:: dict
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/docs/expr/functions.rst]: Fixed this in the import :)

File: python/hail/docs/expr/expression.rst [hunk 2]
```diff
@@ -89,25 +50,10 @@ Expressions
 .. autoclass:: CollectionExpression
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/docs/expr/expression.rst]: not totally sure how to do this right now. lets save for later?

[the remark this answers is not in the document]
Comment (jigold) [reply python/hail/docs/expr/expression.rst]: looks like you added this.

**Comment by jigold** [comment]
Actually, the `unphased_diploid_gt_index_call` is still duplicated. Can you fix?

**Comment by tpoterba** [comment]
Up to you! Happy to do it or defer to you.

File: python/hail/docs/expr/functions/index.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+.. autosummary::
+
+    locus
+    parse_locus
+    parse_variant
+    interval
+    parse_interval
+    call
+    unphased_diploid_gt_index_call
+    unphased_diploid_gt_index_call
```
Comment (jigold) [inline python/hail/docs/expr/functions/index.rst]: remove dup here

**Comment by jigold** [comment]
I can do it.

File: python/hail/docs/expr/functions/index.rst (hunk 3 above)
Comment (tpoterba) [reply python/hail/docs/expr/functions/index.rst]: done

**Pull request merged**